bee-api/bee-rest-api/src/endpoints/routes/api/v2/submit_message.rs

```diff
@@ -122,32 +124,55 @@ pub(crate) async fn submit_message<B: StorageBackend>(
                 .as_str()
                 .ok_or_else(|| {
                     reject::custom(CustomRejection::BadRequest(
-                        "invalid parents: expected an array of message ids".to_string(),
+                        "invalid parent: expected a message id".to_string(),
                     ))
                 })?
                 .parse::<MessageId>()
                 .map_err(|_| {
                     reject::custom(CustomRejection::BadRequest(
-                        "invalid network id: expected an u64-string".to_string(),
+                        "invalid parent: expected a message id".to_string(),
                     ))
                 })?;
             message_ids.push(message_id);
         }
         message_ids
     };
 
-    let payload = if payload_v.is_null() {
+    let payload = if payload_json.is_null() {
         None
     } else {
-        let payload_dto = serde_json::from_value::<PayloadDto>(payload_v.clone())
+        let payload_dto = serde_json::from_value::<PayloadDto>(payload_json.clone())
             .map_err(|e| reject::custom(CustomRejection::BadRequest(e.to_string())))?;
+
+        // TODO: is this check really necessary here considering that the "node processing pipe" anyways checks the
+        // field for incoming messages? In case of a transaction payload, check if its `network_id` matches the
+        // one configured in the node.
```

Comment:
same here

Reply:
Removed the network check as we will rely on the node pipe for this